src/server/gc_worker/gc_worker.rs

```diff
@@ -349,29 +435,39 @@ where
         MvccTxn::new(TimeStamp::zero(), concurrency_manager)
     }
 
-    fn flush_txn(txn: MvccTxn, limiter: &Limiter, engine: &E) -> Result<()> {
+    // Flush the modifications to the disk.
+    // `engine` here represents the storage level engien (ex: RaftKv) which may do
+    // some specific encoding works for each modification.
+    // `kv_engine` here represents the underlying db (ex: RocksEngine)
+    fn flush_txn<EK: KvEngine>(
+        txn: MvccTxn,
+        limiter: &Limiter,
+        engine: &E,
+        kv_engine: EK,
+    ) -> Result<()> {
         let write_size = txn.write_size();
-        let modifies = txn.into_modifies();
+        let mut modifies = txn.into_modifies();
         if !modifies.is_empty() {
             limiter.blocking_consume(write_size);
-            engine.modify_on_kv_engine(modifies)?;
+            engine.adjust_modify(&mut modifies);
+            write_modifies(&kv_engine, modifies)?;
         }
         Ok(())
     }
 
-    fn gc(&mut self, start_key: &[u8], end_key: &[u8], safe_point: TimeStamp) -> Result<()> {
-        if !self.need_gc(start_key, end_key, safe_point) {
+    fn gc(&mut self, store_id: u64, region: Region, safe_point: TimeStamp) -> Result<()> {
```

Comment:
Why not make `store_id` a field of gc runner or gc worker? So you don't need to pass the value around as it never changes once initialized.